app/auth-pages/stores/register.store.ts

```diff
@@ -0,0 +1,91 @@
+import { FormState, Validator } from 'formstate';
+import { action, observable, runInAction } from 'mobx';
+
+import { inject, injectable } from '@servicetitan/react-ioc';
+
+import { Role } from '../../common/models/role';
+import { DropdownFieldState, formStateToJS, InputFieldState } from '../../common/utils/form-helpers';
+import { FormValidators } from '../../common/utils/form-validators';
+import { AuthApi, IsLoginFreeDTO } from '../api/auth.api';
+
+@injectable()
+export class RegisterStore {
+    constructor(
+        @inject(AuthApi)
+        private authApi: AuthApi,
+    ) {
+    }
+
+    @observable
+    registered: boolean = false;
+
+    @observable
+    serverError: string = '';
+
+    strongPasswordValidator: Validator<string> = v =>
+        !FormValidators.passwordIsValidFormat(v)
+        && 'Your password must be at least 8 characters long including a number, ' +
+        'a lowercase letter, and an uppercase letter';
+
+    loginIsNotInUseValidator: Validator<string> = async v =>
+        (await this.isLoginAlreadyInUse(v)) && 'Login is already taken';
+
+    passwordMatchValidator: Validator<string> = v =>
```

Comment:
password and confirm password could be grouped into one FormState to provide a generic validator.